metricshub-wbem-extension/src/main/java/org/sentrysoftware/metricshub/extension/wbem/WbemExtension.java

```diff
@@ -0,0 +1,389 @@
+		return Set.of(WbemSource.class);
+	}
+
+	@Override
+	public Map<Class<? extends IConfiguration>, Set<Class<? extends Source>>> getConfigurationToSourceMapping() {
+		return Map.of(WbemConfiguration.class, Set.of(WbemSource.class));
+	}
+
+	@Override
+	public Set<Class<? extends Criterion>> getSupportedCriteria() {
+		return Set.of(WbemCriterion.class);
+	}
+
+	@Override
+	public void checkProtocol(TelemetryManager telemetryManager) {
+		// Retrieve the hostname
+		final String hostname = telemetryManager.getHostConfiguration().getHostname();
+
+		// Retrieve the host endpoint monitor
+		final Monitor hostMonitor = telemetryManager.getEndpointHostMonitor();
+
+		// Create and set the WBEM result to null
+		String wbemResult = null;
+
+		// Retrieve WBEM configuration from the telemetry manager
+		final WbemConfiguration wbemConfiguration = (WbemConfiguration) telemetryManager
+			.getHostConfiguration()
+			.getConfigurations()
+			.get(WbemConfiguration.class);
+
+		// Stop the WBEM health check if there is not an WBEM configuration
+		if (wbemConfiguration == null) {
+			return;
+		}
+
+		log.info(
+			"Hostname {} - Checking WBEM protocol status. Sending a WQL SELECT request on different namespaces.",
+			hostname
+		);
+
+		final Long strategyTime = telemetryManager.getStrategyTime();
+		final MetricFactory metricFactory = new MetricFactory();
+
+		for (final String wbemNamespace : WBEM_UP_TEST_NAMESPACES) {
+			try {
+				log.info(
+					"Hostname {} - Checking WBEM protocol status. Sending a WQL SELECT request on {} namespace.",
+					hostname,
+					wbemNamespace
+				);
+
+				// The query on the WBEM namespace returned a result
+				//CHECKSTYLE:OFF
+				if (
+					wbemRequestExecutor.executeWbem(
+						hostname,
+						wbemConfiguration,
+						WBEM_TEST_QUERY,
+						wbemNamespace,
+						telemetryManager
+					) !=
+					null
+					//CHECKSTYLE:OFF
+				) {
+					// Collect the metric with a '1.0' value and stop the test
+					metricFactory.collectNumberMetric(hostMonitor, WBEM_UP_METRIC, UP, strategyTime);
+					return;
+				}
+			} catch (Exception e) {
+				if (isAcceptableException(e)) {
+					// Collect the WBEM metric with a '1.0' value and stop the test as the thrown exception is acceptable
+					metricFactory.collectNumberMetric(hostMonitor, WBEM_UP_METRIC, UP, strategyTime);
+					return;
+				}
+				log.debug(
+					"Hostname {} - Checking WBEM protocol status. WBEM exception when performing a WQL SELECT query on '{}' namespace: ",
+					hostname,
+					wbemNamespace,
+					e
+				);
+			}
+		}
+
+		// Generate a metric from the WBEM result
+		// CHECKSTYLE:OFF
+		metricFactory.collectNumberMetric(hostMonitor, WBEM_UP_METRIC, wbemResult != null ? UP : DOWN, strategyTime);
+		// CHECKSTYLE:ON
+	}
+
+	@Override
+	public CriterionTestResult processCriterion(
+		Criterion criterion,
+		String connectorId,
+		TelemetryManager telemetryManager
+	) {
+		if (criterion instanceof WbemCriterion wbemCriterion) {

```

Comment:
I was expecting to see this implementation:
```java
	@Override
	public CriterionTestResult processCriterion(
		Criterion criterion,
		String connectorId,
		TelemetryManager telemetryManager
	) {
		if (criterion instanceof WbemCriterion wbemCriterion) {
			return new WbemCriterionProcessor(wbemRequestExecutor).process(wbemCriterion, connectorId, telemetryManager);
		}
		throw new IllegalArgumentException(
			String.format(
				"Hostname %s - Cannot process criterion %s.",
				telemetryManager.getHostname(),
				criterion != null ? criterion.getClass().getSimpleName() : "<null>"
			)
		);
	}
```

Move all the logic in the WbemCriterionProcessor.